test/functional/JLM_Tests/src/org/openj9/test/java/lang/management/TestOpenJ9DiagnosticsMXBean.java

```diff
@@ -264,6 +265,8 @@ private void testLocal_setDumpOptions() {
 		}		
 
 		try {
+			/* Introduced a sleep to avoid Dump configuration unavailable issue */
+			Thread.sleep(2000);
```

Comment:
As @pdbain-ibm indicated:

> In re 1.) above, I suggest using a mechanism other than a fixed timeout, such as checking for process termination or presence of a file. Fixed timeouts at best slow down the test unnecessarily, and at worst fail intermittently.

I've concerned about the fragility of adding Sleeps to tests to make them pass.  Is there something else we can trigger off?

Reply:
A Sleep is never a "fix".

The PR description indicates:
> As the dump settings are changed one after the other, at times we get a Dump configuration cannot be changed as another dump change is in progress, introducing a delay fixes the issue.

The root question is what else is modifying the dump configuration at this point?

If the exception is expected (!!), then code using the api should be written to loop until the options have been successfully set
```
while(true) {
  try {
    diagBean.setDumpOptions("java:events=catch,filter=java/io/UnsupportedEncodingException,range=1..1,file=" + dumpFilePath);
    break;
  } catch(??? Exception e) {
    // Eat exception and try setting options again
  }
}
```

I hope that's not the expected API usage

Reply:
Unfortunately, I expect that's (close to) the expected usage pattern. I think a (small?) delay is appropriate if `setDumpOptions` throws `DumpConfigurationUnavailableException`.

Reply:
The test code should write that kind of loop than rather than rely on that sleep being sufficient on it's own.

An even better solution may be to reorganize this test to check for success on the first dump before attempting the second.  The creation of the file from the first check could be used a trigger to indicate that its safe to modify the config.

Reply:
I tried both options instead of sleep, the first option worked tried with 50 runs but the second option failed.  

option 1
while(true) {
  try {
    diagBean.setDumpOptions("java:events=catch,filter=java/io/UnsupportedEncodingException,range=1..1,file=" + dumpFilePath);
    break;
  } catch(ConfigurationUnavailableException e) {
    // Eat exception and try setting options again
  }
}

option 2

To check for success on the first dump before attempting the second. The creation of the file from the first check could be used a trigger to indicate that its safe to modify the config.

Reply:
Definitely preferred over brittle timeout (have seen too many times the where timeout values get set, then nudged/increased, because tests are at mercy of machine config, etc).  

In the scenario (however rare) that the config is never available, I guess this while loop hangs the test, but we can/should put upper execution time limits on test suites at the framework level for that case.

Reply:
I don't expect there is a problem to put a short sleep inside the loop used for option 1, in order to give the JVM time to be ready without burning cpu. You could then either count the iterations and fail after some number, or put an upper bound on the execution time as Shelly suggests.